### PR TITLE
377 version param

### PR DESCRIPTION
**Pull request by SofiaSazonova**
### Feature or Bugfix

- Feature

### Detail
version.json in repo root contains information about version. This json-string is uploaded to SSM parameter /dataall/{env}/versin. Also this file is copied to frontend folder and version number is displayed near "dataall" header 

### Relates
https://github.com/data-dot-all/dataall/issues/377

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)? NA
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization? NA
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features? NA
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?NA
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by SofiaSazonova** [comment]
As long, as @dlpzx required simple way to store version info, I made it as obvious as possible.
Version info is updated only in one file - version.json
We will both have ssm-param for that and static file in the front.

**What can be different?**
We can introduce another API route (smth like /metadata) to fetch version from backend and display at the frontend.
In this case it will be 100% the same at front and back, but maybe in future we want to have different versions for front and back? Also, I decided not to make another API section just for one version-param.

**Comment by SofiaSazonova** [comment]
Screenshot 2024-01-19 at 20 01 48
Screenshot 2024-01-19 at 20 02 05
Screenshot 2024-01-19 at 20 03 00

File: deploy/stacks/pipeline.py [hunk 1]
```diff
@@ -558,6 +559,7 @@ def set_quality_gate_stage(self):
                         'mkdir -p source_build',
                         'mv backend ./source_build/',
                         'mv config.json ./source_build/',
+                        'mv version.json ./source_build/',
```
Comment (noah-paige) [inline deploy/stacks/pipeline.py:562]: Do we need this line for `UploadCodeToS3`? I think we can remove as the `source_build.zip` is used primarily for DBMigration step only which should not be dependent on version

**Comment by noah-paige** [comment]
Left a minor comment - otherwise code looks good and a very much needed enhancement 🙌 

To finish up testing and push through by Monday:

Deployed to AWS (Internet Facing)
- [ ] CICD Successful
- [ ] Version present in login logo + webpage

Deployed to AWS (VPC Facing)
- [ ] CICD Successful
- [ ] Version present in login logo + webpage

Deployed Local
- [ ] Logo present in webpage

**Comment by SofiaSazonova** [comment]
Now version is tooltip
image

File: deploy/stacks/pipeline.py (hunk 1 above)
Comment (noah-paige) [reply deploy/stacks/pipeline.py:562]: Let's remove this line `mv version.json ./source_build/` - I tested to ensure it is not needed

I think best to keep hardcoding of `version.json` minimal to only when necessary

Will approve right after this change

Comment (SofiaSazonova) [reply deploy/stacks/pipeline.py]: Done

**Review by noah-paige — APPROVED** [review]
Looks good - approving!

**Pull request merged**